api/application_submissions/index.js

```diff
@@ -0,0 +1,65 @@
+const express = require("express");
+
+const router = express.Router();
+
+const ApplicationSubmission = require("../../models/applicationSubmissionsModel");
+
+
+router.get("/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+    const submission = await ApplicationSubmission.findById(id);
+    if (submission) {
+      res.status(200).json({ submission, error: null });
+    } else {
+      res.status(400).json({ message: "Submission not found in the database" });
+    }
+  } catch (error) {
+    res.status(500).json({ error, message: "Unable to make request to server" });
+  }
+});
+
+router.post("/", async (req, res) => {
+  const postSub = {
+    ...req.body,
+    decision: "PENDING"
+  };
+  try {
+    const [newSubmission] = await ApplicationSubmission.insert(postSub);
+    if (newSubmission) {
+      res.status(201).json({ newSubmission, message: "Submission added to database" });
+    }
+  } catch (error) {
+    res.status(500).json({ error, message: "Unable to add submission" });
+  }
+});
+
+router.put("/:id", async (req, res) => {
+  const { id } = req.params;
+  const { decision } = req.body;
+
+  try {
+    // update all other submissions for that skilled impact request to denied
+    const applicationSubmission = await ApplicationSubmission.findById(id);
+
+    if(applicationSubmission) {
+      const skilled_impact_request_id = applicationSubmission["skilled_impact_request_id"];
+      const allSkilledImpactRequestSubmissions = 
+            await ApplicationSubmission.findAllBySkilledImpactRequestId(skilled_impact_request_id);
+      
+      for(let i = 0; i < allSkilledImpactRequestSubmissions.length; i++) {
+        await ApplicationSubmission.update(allSkilledImpactRequestSubmissions[i].id, "DENIED");
+      }
+      // now, update current submission
+      const [updatedSubmission] = await ApplicationSubmission.update(id, decision);
```

Comment:
If you're setting all the applications for a skilled impact request to "DENIED" and then updating the current application to its correct decision, there's a chance that something goes wrong after you set the current application to "DENIED" but before you update it to the correct decision, and then you gotta be super careful about making a bunch of related database calls.

If you want to avoid some of the database calls going through and some of the calls not going through, use database transactions (simple explanation [here](https://stackoverflow.com/questions/974596/what-is-a-database-transaction)) to make sure they either all go through or don't.